<a href="https://colab.research.google.com/github/JordanDouglasRDM/Conhecendo_Python/blob/main/Jordan_Douglas_221_Est_DecN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Lista de exercícios de ESTRUTURA de DECISÃO
#Fazer do 1 ao 25, o restante fica para estudos

### 1.a) Faça um algoritmo que calcule e exiba o salário reajustado de um funcionário de acordo com a seguinte regra: Salário até 500, reajuste de 50%.

In [ ]:
salario = float(input('Informe o salário: '))
if salario <= 500:
    novo = salario + (salario * 50 / 100)
    print('O salário reajustado é',novo)

Informe o salário: 408
O salário reajustado é 612.0


###1.b) Faça um algoritmo que calcule e exiba o salário reajustado de um funcionário de acordo com a seguinte regra: Salário até 500, reajuste de 50%; Salários maiores que 500, reajuste de 30%.

In [ ]:
salario = float(input('Informe o salário: '))
if salario <= 500:
    novo = salario + (salario * 50 / 100)
    print('O salário reajustado é ',novo, 'R$')
else: 
    novo = salario + (salario * 30 / 100)
    print('O salário reajustado é', novo, 'R$')

Informe o salário: 600
O salário reajustado é 780.0 R$


###2. Faça um algoritmo que leia o nome e a idade de uma pessoas, verifique se a idade de uma pessoa é menor ou maior de idade. Considera-se maior de idade uma pessoa com 18 anos ou mais. Como saída o algoritmo deve informar o nome e a idade da pessoa e depois uma mensagem se ela é ou não maior de idade.


In [ ]:
nome = str(input('Informe o seu nome: '))
idade = int(input('Informe sua idade: '))

if idade >= 18:
  maior_menor = 'maior de idade'
else:
  maior_menor = 'menor de idade'
print('O seu nome é ',nome, 'e você tem ',idade, 'anos.' )
print('Você é ', maior_menor)

Informe o seu nome: jr
Informe sua idade: 17
O seu nome é  jr e você tem  17 anos.
Você é  menor de idade


###3.a) Faça um algoritmo que leia duas notas de um aluno, calcule a média e verifique, apresentando, se está aprovado ou reprovado.

In [ ]:
nota1 = float(input('Informe a nota da P1: '))
nota2 = float(input('Informe a nota da P2: '))
media = (nota1 + nota2) / 2
if media >= 6:
  print('Você foi aprovado')
else:
  print('Você foi reprovado')


Informe a nota da P1: 8.5
Informe a nota da P2: 9.5
Você foi aprovado


###3.b) Faça um algoritmo que calcule e apresente o que foi requerido no exercício 3.a) e também avalie a condição de aprovar/reprovar, apenas quando o aluno tem frequencia acima de 75%, este valor deve ser lido.

In [ ]:
nota1 = float(input('Informe a nota da P1: '))
nota2 = float(input('Informe a nota da P2: '))
fre = float(input('Informe a frequência do aluno: '))
media = (nota1 + nota2) / 2
if media >= 6 and fre >= 75:
  print('Você foi aprovado')
else:
  print('Você foi reprovado')

Informe a nota da P1: 7
Informe a nota da P2: 8
Informe a frequência do aluno: 76
Você foi aprovado


###4. Faça um algoritmo que leia o gênero de uma pessoa. Se for digitado M ou F, apresentar 'Gênero válido!'. Caso contrário, 'Gênero inválido!'.

In [ ]:
gen = str(input('Informe o seu gênero (m para masculino e f para feminino): '))
if gen == "m" or gen == "f":
    print('Gênero válido!')
else:
    print('Gênero inválido!')

Informe o seu gênero (m para masculino e f para feminino): g
Gênero inválido!


###5. Tendo como dados de entrada a altura e o gênero (M/F) de uma pessoa (M-masculino ou F-feminino), construa um algoritmo que calcule seu peso ideal, utilizando as seguintes fórmulas:
masculino: (72.7 * altura) - 58;

feminino: (62.1 * altura) - 44.7



In [ ]:
gen = (input('Informe o seu gênero (M para masculino e F para feminino): '))
altura = float(input('Informe sua altura: '))
if gen == "M":
  peso = (72.7 * altura) - 58
  print(F'Você é {gen} e seu peso ideal é {peso:.2f}')
elif gen == "F":
  peso = (62.1 * altura) - 44.1
  print(F'Você é {gen} e seu peso ideal é {peso:.2f} kg')

Informe o seu gênero (M para masculino e F para feminino): M
Informe sua altura: 1.81
Você é M e seu peso ideal é 73.59


###6. Faça um algoritmo que leia dois números inteiros e mostre o resultado da **diferença** do maior valor pelo menor.

In [ ]:
num1 = int(input('Informe um valor para o número 1: '))
num2 = int(input('Informe um valor para o número 2: '))
if num1 > num2:
    maior = (num1)
    menor = (num2)
else:
    maior = (num2)
    menor = (num1)  
dif = (maior - menor)
print('A difereça do maior número pelo menor número é: ', dif)

Informe um valor para o número 1: 89
Informe um valor para o número 2: 15
A difereça do maior número pelo menor número é:  74


###7. Uma faculdade faz o pagamento de seus professores por hora/aula. Faça um algoritmo que receba o nível e a quantidade de hora/aula, calcule e exiba o salário de um professor e a frase a seguira tabela abaixo:
Professor Nível 1 - 11,00 reais por hora/aula, salário: ...

Professor Nível 2 - 15,00 reais por hora/aula, salário: ...

Professor Nível 3 - 19,00 reais por hora/aula, salário: ...

In [ ]:
nivel = int(input('Informe qual o nível do professor: '))
hora_aula = float(input('Informe a quantidade de hora/aula: '))
if nivel == 1:
    salario = (hora_aula * 11)
elif nivel == 2:
    salario = (hora_aula * 15)
elif nivel == 3:
    salario = (hora_aula * 19)
print('Professor Nível', nivel, '- 11,00R$ por hora/aula, salário: R$', salario)

Informe qual o nível do professor: 3
Informe a quantidade de hora/aula: 40
Professor Nível 3 - 11,00R$ por hora/aula, salário: R$ 760.0


###8. Faça um algoritmo que leia um número inteiro e mostre uma mensagem indicando se este número é par ou ímpar e se é positivo ou negativo.


In [ ]:
num = int(input('Informe um número pertencente ao conjunto dos Inteiros: '))
resto = num % 2
if resto == 0:
  impar_par = 'par'
else:
  impar_par = 'impar'
if num >= 0:
  x = 'positivo'
else:
  x = 'negativo'
print('O número informado é',impar_par, 'e é um número',x)

Informe um número pertencente ao conjunto dos Inteiros: -897
O número informado é impar e é um número negativo


###9. Num determinado Estado, para transferências de veículos, o DETRAN cobra uma taxa de 1% para carros fabricados antes de 1990 e uma taxa de 1.5% para os fabricados de 1990 em diante, taxa esta incidindo sobre o valor de tabela do carro. Faça um algoritmo que leia o ano e o preço do carro, calcule e apresente o imposto a ser pago.

In [ ]:
valor = float(input('Informe o valor do veículo: '))
ano = int(input('Informe o ano de fabricação: '))
if ano < 1990:
  valor = (valor * 0.01)
  print('O valor do imposto a ser pago é de R$',valor)
else:
  valor = (valor * 0.015)
  print('O valor do imposto a ser pago é de R$',valor)

Informe o valor do veículo: 158000
Informe o ano de fabricação: 2022
O valor do imposto a ser pago é de R$ 2370.0


###10. Elabore um algoritmo que informando a idade de um nadador o mesmo terá condições de classificar em uma das seguintes categorias: 
infantil = 5 - 10 anos;
 
juvenil = 11-17 anos; 

adulto = maiores de 18 anos.



In [ ]:
idade = int(input('Informe a idade do nadador: '))
if idade >= 5 and idade <= 10:
  print('O nadador poderá se classificar na categoria infantil')
elif idade >= 11 and idade <= 17:
  print('O nadador poderá se classificar na categoria juvenil')
elif idade >= 18:
  print('O nadador poderá se classificar na categoria adulto')
elif idade < 5:
  print('O nadador não tem idade suficiente para se classificar')

Informe a idade do nadador: 4
O nadador não tem idade suficiente para se classificar


###11. Um banco concederá um crédito especial aos seus clientes, variável com o saldo médio no último ano. Faça um algoritmo que leia o saldo médio de um cliente e calcule o valor do crédito de acordo com a tabela a seguir. Mostre uma mensagem informando o saldo médio e o valor do crédito.
        Saldo médio
        * de 0 a 200 nenhum crédito
        * de 201 a  400 20% do valor do saldo médio
        * de 401 a  600 30% do valor do saldo médio
        * acima de 601 40% do valor do saldo médio


In [ ]:
saldo_medio = int(input('Informe o valor do saldo médio: '))
if saldo_medio <= 200:
  print('Cliente não esstá dentro dos parâmetros para receber crédito!')
elif saldo_medio <= 400 and saldo_medio > 200:
  valor_cred = saldo_medio * 0.2
  print('O valor do saldo médio é', saldo_medio,'e está disponível',valor_cred,'R$ de crédito.' )
elif saldo_medio <= 600 and saldo_medio > 400:
  valor_cred = saldo_medio * 0.3
  print('O valor do saldo médio é', saldo_medio,'e está disponível',valor_cred,'R$ de crédito.' )
elif saldo_medio > 600:
  valor_cred = saldo_medio * 0.4
  print('O valor do saldo médio é', saldo_medio,'e está disponível',valor_cred,'R$ de crédito.' )

Informe o valor do saldo médio: 250
O valor do saldo médio é 250 e está disponível 50.0 R$ de crédito.


###12. A Organização Mundial de Saúde usa a seguinte tabela para determinar a condição de um adulto, para isso desenvolva um algoritmo para calcular o Índice de Massa Corporal (IMC) e apresenta-lo, dado pela fórmula: 
    IMC = peso / (altura)2   (o número 2 significa, elevado ao quadrado)

CONDIÇÃO |  IMC em adultos |
 ---|---
Abaixo do peso  |	Abaixo de 18.5  |
No peso normal	|	Entre 18.5 e 25|
Acima do peso	|	Entre 25.1 e 30|
Obeso		    |	Acima de 30|

In [ ]:
altura = float(input('Informe sua altura: '))
peso = float(input('Informe seu peso: '))
imc = peso / (altura ** 2)
if imc < 18.5:
  print('Você está abaixo do peso!')
elif imc >= 18.5 and imc <= 25:
  print('Você está no peso ideal!')
elif imc > 25 and imc <= 30:
  print('Você está acima do peso!')
elif imc > 30:
    print('Você é obeso!')

Informe sua altura: 1.55
Informe seu peso: 60
Você está no peso ideal!


###13. A nota final de um estudante é calculada a partir de três notas atribuídas, respectivamente, a um trabalho de laboratório, a uma avaliação semestral e a um exame final. A média das três notas mencionadas obedece aos pesos a seguir:

 NOTA |  PESO |
 ---|---
Trabalho de laboratório  |   2   |
Avaliação semestral      |   3   |
Exame final              |   5   |

### Faça um programa que receba as três notas, calcule e mostre a média ponderada e o conceito que segue a tabela

MÉDIA PONDERADA    | CONCEITO|
---|---
8,0 <= média <= 10  |    A 
7,0 <= média < 8,0  |    B 
6,0 <= média < 7,0  |    C 
5,0 <= média < 6,0  |    D 
 0,0 <= média < 5,0  |    E
       
media_ponderada = (nota_traballho * 2 + avaliacao_semestral * 3 + exame_final * 5) / 10



In [ ]:
trab_lab = float(input('Informe a nota do Trabalho do Laboratório: '))
ava_sem = float(input('Informe a nota da Avaliação Semestral: '))
exa_fin = float(input('Informe a nota do Exame Final: '))
med_pond = ((trab_lab * 2) + (ava_sem * 3) + (exa_fin * 5)) / 10
if 0 <= med_pond < 5:
  conc = 'E'
  print('Sua média foi ', med_pond, 'e seu conceito foi ',conc )
elif 5 <= med_pond < 6:
  conc = 'D'
  print('Sua média foi ', med_pond, 'e seu conceito foi ',conc )
elif 6 <= med_pond < 7:
  conc = 'C'
  print('Sua média foi ', med_pond, 'e seu conceito foi ',conc )
elif 7 <= med_pond < 8:
  conc = 'B'
  print('Sua média foi ', med_pond, 'e seu conceito foi ',conc )
elif 8 <= med_pond < 10:
  conc = 'A'
  print('Sua média foi ', med_pond, 'e seu conceito foi ',conc )


Informe a nota do Trabalho do Laboratório: 11
Informe a nota da Avaliação Semestral: 10
Informe a nota do Exame Final: 8
Sua média foi  9.2 e seu conceito foi  A


### 14. Faça um programa que receba três notas de um aluno, calcule e mostre a média aritmética e a mensagem constante na tabela a seguir. Aos alunos que ficaram para exame, calcule e mostre a nota que deverão tirar para serem aprovados, considerando que a média exigida é 6,0.

MÉDIA ARITMÉTICA | MENSAGEM
    ---|---
    0 <= média < 3 |Reprovado
    3 <= média < 6|Exame
    6 <= média <= 10|Aprovado

```
SE media >= 3 E media < 6
    ESCREVA “Exame”
    nota_exame = 12 - media
    ESCREVA “Você deve tirar a nota”, nota_exame, “para ser aprovado.”


In [ ]:
p1 = float(input('Informe a nota da P1: '))
p2 = float(input('Informe a nota da P2: '))
p3 = float(input('Informe a nota da P3: '))
med = (p1 + p2 + p3) / 3
if 0 <= med < 3:
  print('Sua média é',med,'e você foi REPROVADO!')
elif 3 <= med < 6:
  exa = 12 - med)
  print('Você está de EXAME e precisa tirar', exa, 'no exame para ser aprovado.')
elif 6 <= med <= 10:
  print('Sua média é',med,'e você foi APROVADO!')

Informe a nota da P1: 3.5
Informe a nota da P2: 3.5
Informe a nota da P3: 3.5
Você está de EXAME e precisa tirar 8 no exame para ser aprovado.


### 15. Faça um programa que receba dois números e mostre o maior.


In [ ]:
A = float(input('Informe um valor para A: '))
B = float(input('Informe um valor para B: '))
if A > B:
  maior = A
  print('O maior número é',maior)
elif B > A:
  maior = B
  print('O maior número é',maior)
elif A == B:
  print('Impossível identificar um número maior, pois ambos são iguais!')

Informe um valor para A: 5
Informe um valor para B: 5
Impossível identificar um número maior, pois ambos são iguais!


### 16. Faça um programa que receba três números e mostre-os em ordem crescente. Suponha que o usuário digitará três números diferentes.

```
SE numero1 < numero2 E numero1 < numero3
    SE numero2 < numero3
        ESCREVA “A ordem crescente é: “,numero1,“-”,numero2,“-”,numero3
    SENÃO
        ESCREVA “A ordem crescente é: “,numero1,“-”,numero3,“-”,numero2
SE numero2 < numero1 E numero2 < numero3
    SE numero1 < numero3
        ESCREVA “A ordem crescente é: “,numero2,“-”,numero1,“-”,numero3
    SENÃO
        ESCREVA “A ordem crescente é: “,numero2,“-”,numero3,“-”,numero1
SE numero3 < numero1 E numero3 < num2
    SE numero1 < numero2
        ESCREVA “A ordem crescente é: “,numero3,“-”,numero1,“-”,numero2
    SENÃO 
        ESCREVA “A ordem crescente é: “,numero3,“-”,numero2,“-”,numero1
  ```

In [ ]:
a = int(input('Informe um número para A: '))
b = int(input('Informe um número para B: '))
c = int(input('Informe um número para C: '))
if a > b and a > c:
    if b > c:
        print(F'A ordem crescente é {c} {b} {a}')
    else:
        print(F'A ordem crescente é {b} {c} {a}')
elif b > a and b > c:
    if a > c:
        print(F'A ordem crescente é {c} {a} {b}')
    else:
        print(F'A ordem crescente é {a} {c} {b}')
elif c > a and c > b:
    if a > b:
        print(F'A ordem crescente é {b} {a} {c}')
    else:
        print(F'A ordem crescente é {a} {b} {c}')

Informe um número para A: 15
Informe um número para B: 13
Informe um número para C: 5
a ordem crescente é:  5 13 15


### 17. Faça um programa que receba três números obrigatoriamente em ordem crescente e um quarto número que não siga essa regra. Mostre, em seguida, os quatro números em ordem decrescente. Suponha que o usuário digitará quatro números diferentes.

```code
SE numero4 > numero3
    ESCREVA “A ordem decrescente é: “,numero4,“-”,numero3,“-”,numero2,“-”,numero1
SENÃO SE numero4 > numero2 E numero4 < numero3
    ESCREVA “A ordem decrescente é: “,numero3,“-”,numero4,“-”,numero2,“-”,numero1
SENÃO SE numero4 > numero1 E numero4 < numero2
    ESCREVA “A ordem decrescente é: “,numero3,“-”,numero2,“-”,numero4, “-”,numero1
SENÃO SE numero4 < numero1
    ESCREVA “A ordem decrescente é: “,numero3,“-”,numero2,“-”,numero1,“-”,numero4
```

In [ ]:
a = int(input('Informe um número para A: '))
b = int(input('Informe um número para B: '))
c = int(input('Informe um número para C: '))
d = int(input('Informe um número para D: '))
if d > c:
    print(F'A ordem decrescente é {d} {c} {b} {a}')
elif d < c and d > b:
    print(F'A ordem decrescente é {c} {d} {b} {a}')
elif d < b and d > a:
    print(F'A ordem decrescente é {c} {b} {d} {a}')
else:
    print(F'A ordem decrescente é {c} {b} {a} {d}')

Informe um número para A: 2
Informe um número para B: 5
Informe um número para C: 8
Informe um número para D: 9
A ordem decrescente é 9 8 5 2


### 18. Faça um programa que receba quatro valores: I, A, B e C. Desses valores, I é inteiro e positivo, A, B e C são reais. Escreva os números A, B e C obedecendo à tabela a seguir.

*Suponha que o valor digitado para I seja sempre um valor válido, ou seja, 1, 2 ou 3, e que os números digitados sejam diferentes um do outro.*

  |VALOR DE I | FORMA A ESCREVER|
  |---|---|
  |1                  |A, B e C em ordem crescente.|
  |2                  |A, B e C em ordem decrescente|
  |3                  |O maior fica entre os outros dois números.|

  ```code
SE I == 1
    SE A<B E A<C
        SE B<C
            ESCREVA “A ordem crescente dos números é:”,A,” -”,B,”-”,C
        SENÃO
            ESCREVA “A ordem crescente dos números é:”,A,” -”,C,”-”,B
    SE B<A E B<C
        SE A<C
            ESCREVA “A ordem crescente dos números é:”,B,”-”,A,”-”,C
        SENÃO
            ESCREVA “A ordem crescente dos números é: “,B,”-”,C,”-”,A
    SE C<A E C<B
        SE A<B
            ESCREVA “A ordem crescente dos números é: “,C,”-”,A,”-”,B
        SENÃO
            ESCREVA “A ordem crescente dos números é: “,C,”-”,B,”-”,A
SE I == 2
    SE A>B E A>C
        SE B>C
            ESCREVA “A ordem decrescente dos números é: “,A,” -”,B,”-”,C
        SENÃO
            ESCREVA “A ordem decrescente dos números é: “,A,” -”,C,”-”,B
    SE B>A E B>C
        SE A>C
            ESCREVA “A ordem decrescente dos números é: “,B,” -”,A,”-”,C
        SENÃO
            ESCREVA “A ordem decrescente dos números é: “,B,” -”,C,”-”,A
    SE C>A E C>B
        SE A>B
            ESCREVA “A ordem decrescente dos números é: “,C,” -”,A,”-”,B
        SENÃO
            ESCREVA “A ordem decrescente dos números é: “,C,” -”,B,”-”,A
SE I == 3
    SE A>B E A>C
        ESCREVA “A ordem desejada é: “,B,”-”,A,”-”,C
    SE B>A E B>C
        ESCREVA “A ordem desejada é: “,A,”-”,B,”-”,C
    SE C>A E C>B
        ESCREVA “A ordem desejada é: “,A,”-”,C,”-”,B
```

In [ ]:
i = int(input('Informe um número para I: '))
a = float(input('Informe um número para A: '))
b = float(input('Informe um número para B: '))
c = float(input('Informe um número para C: '))
if i == 1:
    if a > b and a > c:
        if b > c:
            print(F'A ordem crescente é {c} {b} {a}')
        else:
            print(F'A ordem crescente é {b} {c} {a}')
    elif b > a and b > c:
        if a > c:
            print(F'A ordem crescente é {c} {a} {b}')
        else:
            print(F'A ordem crescente é {a} {c} {b}')
    elif c > a and c > b:
        if a > b:
            print(F'A ordem crescente é {b} {a} {c}')
        else:
            print(F'A ordem crescente é {a} {b} {c}')
elif i == 2:
    if a < b and a < c:
        if b < c:
            print(F'A ordem decrescente é {c} {b} {a}')
        else:
            print(F'A ordem decrescente é {b} {c} {a}')
    elif b < a and b < c:
        if a < c:
            print(F'A ordem decrescente é {c} {a} {b}')
        else:
            print(F'A ordem decrescente é {a} {c} {b}')
    elif c < a and c < b:
        if a < b:
            print(F'A ordem decrescente é {b} {a} {c}')
        else:
            print(F'A ordem decrescente é {a} {b} {c}')
elif i == 3:
    if a > b and a > c:
        print(F'Escrevendo com o maior número no meio, fica {b} {a} {c}')
    elif b > a and b > c:
        print(F'Escrevendo com o maior número no meio, fica {a} {b} {c}')
    elif c > b and c > a:
        print(F'Escrevendo com o maior número no meio, fica {b} {c} {a}')

Informe um número para I: 1
Informe um número para A: 375
Informe um número para B: 951
Informe um número para C: 3587
A ordem crescente é 375.0 951.0 3587.0


### 19. Faça um programa que mostre o menu de opções a seguir, receba a opção do usuário, número 1 para a escolha da soma, ou número 2 para a raiz quadrada, também receba os dados necessários para executar cada operação.

    Menu de opções:
    1. Somar dois números.
    2. Raiz quadrada de um número.

Observação: raiz = numero ** (1/2), ou 
import math
raiz = math.sqrt(numero)
      

In [ ]:
print('Menu de opções:')
print('1. Somar dois números.')
print('2. Raiz quadrada de um número.')
opc = int(input('Qual opção você deseja? '))
if opc == 1:
    num1 = int(input('Informe um valor para o primeiro número: '))
    num2 = int(input('Informe um valor para o segundo número: '))
    print(F'A soma dos números é {num1 + num2}')
elif opc == 2:
    num1 = float(input('Informe o número que deseja calcular a raiz quadrada: '))
    import math
    raiz = float(math.sqrt(num1))
    print(F'A raiz quadrada de {num1} é {raiz:.3f}')
else:
    print('Você não informou um número válido no **menu de opções**, por favor, reinicie o programa e tente novamente!')

Menu de opções:
1. Somar dois números.
2. Raiz quadrada de um número.
Qual opção você deseja? 2
Informe o número que deseja calcular a raiz quadrada: 98456
A raiz quadrada de 98456.0 é 313.777


### 20. Faça um programa que mostre a data e a hora do sistema nos seguintes formatos: **dia/mês/ano – mês por extenso e hora:minuto**.

Observação: ***from*** carrega um módulo/biblioteca da linguagem Python e o ***import*** é usado para informar qual objeto desta biblioteca queremos importar/carregar no nosso programa

    from datetime import datetime
    #captura a data e hora de hoje de algum país onde esta o servidor que executa nosso programa em Python
    #quando eu testei, o servidor era de Dacar-Senegal
    hoje = datetime.now()
    print ('Ano atual.:',hoje.year)
    print ('Mês.......:',hoje.month)
    print ('Dia.......:',hoje.day)
    print ('Hora......:',hoje.hour)
    print ('Minuto....:',hoje.minute)
    print ('Segundos..:',hoje.second)

In [ ]:
from datetime import datetime
hoje = datetime.now()
print ('Ano atual.:',hoje.year)
print ('Mês.......:',hoje.month)
print ('Dia.......:',hoje.day)
print ('Hora......:',hoje.hour)
print ('Minuto....:',hoje.minute)
print ('Segundos..:',hoje.second)
if hoje.month == 3:
    print(hoje.day,'/Janeiro/',hoje.year)
...
...

Ano atual.: 2022
Mês.......: 4
Dia.......: 1
Hora......: 12
Minuto....: 1
Segundos..: 3


Ellipsis

### 21. Faça um programa que determine a data cronologicamente maior entre duas datas fornecidas pelo usuário. Cada data deve ser composta por três valores inteiros, em que o primeiro representa o dia, o segundo, o mês e o terceiro, o ano.

  ```code
  SE ano1 > ano2
      ESCREVA “A data mais antigaantiga é: “,dia1,”-”,mes1,”-”,ano1
  SENÃO SE ano2>ano1
      ESCREVA “A data mais antiga é: “,dia2,”-”,mes2,”-”,ano2
  SENÃO SE mes1>mes2
      ESCREVA “A data mais antiga é: “,dia1,”-”,mes1,”-”,ano1
  SENÃO SE mes2>mes1
      ESCREVA “A data mais antiga é: “,dia2, “-”,mes2,”-”,ano2
  SENÃO SE dia1>dia2
      ESCREVA “A data mais antiga é: “-”,dia1,”-”,mia1,” -”,ano1
  SENÃO SE dia2>dia1
      ESCREVA “A data mais antiga é: “,dia2,” -”,mes2,”-”,ano2
  SENÃO
      ESCREVA “As datas são iguais!”
  ```

In [ ]:
year1 = int(input('Informe o ano da primeira data (com 4 digitos): '))
year2 = int(input('Informe o ano da segunda data (com 4 digitos): '))
month1 = int(input('Informe o mês da primeira data (número entre 1 e 12): '))
month2 = int(input('Informe o mês da segunda data (número entre 1 e 12): '))
day1 = int(input('Informe o dia da primeira data (número entre 1 e 31: '))
day2 = int(input('Informe o dia da segunda data (número entre 1 e 31: '))

if day1 >= 32 or day2>= 32 or month1 >= 13 or month2 >= 13:
    print('Dados inseridos incorretamente!')
else:
    if year1 != year2:
        if year1 < year2:
            print(F'A data mais antiga é: {day1}/{month1}/{year1}')
        else:
            print(F'A data mais antiga é: {day2}/{month2}/{year2}')
    elif year1 == year2:
        if month1 != month2:
            if month1 < month2:
                print(F'A data mais antiga é: {day1}/{month1}/{year1}')
            else:
                print(F'A data mais antiga é: {day2}/{month2}/{year2}')
        elif month1 == month2:
            if day1 != day2:
                if day1 < day2:
                    print(F'A data mais antiga é: {day1}/{month1}/{year1}')
                else:
                    print(F'A data mais antiga é: {day2}/{month2}/{year2}')
            else:
                print('As datas informadas são iguais!')

Informe o ano da primeira data (com 4 digitos): 2000
Informe o ano da segunda data (com 4 digitos): 2000
Informe o mês da primeira data (número entre 1 e 12): 5
Informe o mês da segunda data (número entre 1 e 12): 6
Informe o dia da primeira data (número entre 1 e 31: 16
Informe o dia da segunda data (número entre 1 e 31: 14
A data mais antiga é: 16/5/2000


### 22. Faça um programa que receba a hora do início de um jogo e a hora do término (cada hora é composta por duas variáveis inteiras: hora e minuto). Calcule e mostre a duração do jogo (horas e minutos), sabendo que o tempo máximo de duração do jogo é de 24 horas e que ele pode começar em um dia e terminar no dia seguinte.
    SE min_inicial > min_f
        minuto_final = minuto_final + 60
        hora_final = hora_final – 1
    SE hora_inicial > hora_final
        hora_final = hora_final + 24
    minuto_duracao = minuto_final - minuto_inicial
    hora_duracao = hora_final - hora_inicial

In [ ]:
h_ini = int(input('Informe a hora do início do jogo: '))
m_ini = int(input('Informe os minutos do início do jogo: '))
h_term = int(input('Informe a hora do término do jogo: '))
m_term = int(input('Informe os minutos do término do jogo: '))
dur_hr = 0
dur_min = 0
if  h_term > h_ini:
    dur_hr = h_term - h_ini
elif h_ini > h_term:
    dur_hr = h_ini - h_term
if  m_term > m_ini:
        dur_min = m_term - m_ini
elif m_ini > m_term:
        dur_min = m_ini - m_term
print(F'O jogo teve de duração: {dur_hr} horas e {dur_min} minutos.')

Informe a hora do início do jogo: 21
Informe os minutos do início do jogo: 45
Informe a hora do término do jogo: 22
Informe os minutos do término do jogo: 55
O jogo teve de duração: 1 horas e 10 minutos.


### 23. Faça um programa que receba o código correspondente ao cargo de um funcionário e seu salário atual e mostre o cargo, o valor do aumento e seu novo salário. Os cargos estão na tabela a seguir.

  |CÓDIGO|CARGO|PERCENTUAL|
  |---|---|---|
  |1|Escriturário|50%|
  |2|Secretário|35%|
  |3|Caixa|20%|
  |4|Gerente|10%|
  |5|Diretor|Não tem aumento|

```code
SE cargo == 1
    ESCREVA “O cargo é Escriturário”
    aumento = salario * 50 / 100
    ESCREVA “O valor do aumento é: “, aumento
    novo_sal = salario + aumento
    ESCREVA “O novo salário é: “, novo_sal
SENÃO SE cargo == 2
    ESCREVA “O cargo é Secretário”
    aumento = salario * 35 / 100
    ESCREVA “O valor do aumento é: “, aumento
    novo_sal = salario + aumento
    ESCREVA “O novo salário é: “, novo_sal]
SENÃO SE cargo == 3
    ESCREVA “O cargo é Caixa”
    aumento = salario * 20 / 100
    ESCREVA “O valor do aumento é: “, aumento
    novo_sal = salario + aumento
    ESCREVA “O novo salário é: “,novo_sal
SENÃO SE cargo == 4
    ESCREVA “O cargo é Gerente”
    aumento = salario * 10 / 100
    ESCREVA “O valor do aumento é: “, aumento
    novo_sal = salario + aumento
    ESCREVA “O novo salário é: “, novo_sal
SENÃO SE cargo == 5
    ESCREVA “O cargo é Diretor”
    aumento = salario * 0 / 100
    ESCREVA “O valor do aumento é: “, aumento
    novo_sal = salario + aumento
    ESCREVA “O novo salário é: “, novo_sal
```

In [ ]:
ccargo = int(input('Inform o código do cargo: '))
salarioa = float(input('Informe o salário atual: '))
if ccargo == 1:
    cargo = "Escrituário"
    aumento = salarioa * 0.5
    nsalario = salarioa + aumento
elif ccargo == 2:
    cargo = "Secretário"
    aumento = salarioa * 0.35
    nsalario = salarioa + aumento
elif ccargo == 3:
    cargo = "Caixa"
    aumento = salarioa * 0.2
    nsalario = salarioa + aumento
elif ccargo == 4:
    cargo = "Gerente"
    aumento = salarioa * 0.1
    nsalario = salarioa + aumento
elif ccargo == 5:
    cargo = "Diretor"
    aumento = 0
    nsalario = salarioa
print(F'O cargo é {cargo} com aumento de R${aumento} e o novo salario é R${nsalario}')

Inform o código do cargo: 5
Informe o salário atual: 1830
O cargo é Diretor com aumento de R$0 e o novo salario é R$1830.0


### 24. Faça um programa que apresente o menu a seguir, permita ao usuário escolher a opção desejada, receba os dados necessários para executar a operação e mostre o resultado. Verifique a possibilidade de opção inválida e não se preocupe com restrições, como salário negativo.

    Menu de opções
    1. Imposto
    2. Novo salário
    3. Classificação
    Digite a opção desejada:

*Na opção 1: receber o salário de um funcionário, calcular e mostrar o valor do imposto usando as regras a seguir.*

  |SALÁRIO|PERCENTUAL DO IMPOSTO|
  |---|---|
  |Menor que  500,00|5%|
  |De  500,00 (inclusive) a  850,00 (inclusive)|10%|
  |Acima de  850,00|15%|

*Na opção 2: receber o salário de um funcionário, calcular e mostrar o valor do novo salário, usando as regras a seguir.*

  |SALÁRIO|AUMENTO|
  |---|---|
  |Maior que  1.500,00| 25,00|
  |De  750,00 (inclusive) a  1.500,00 (inclusive)| 50,00|
  |De  450,00 (inclusive) a  750,00| 75,00|
  |Menor que  450,00| 100,00|

*Na opção 3: receber o salário de um funcionário e mostrar sua classificação usando a tabela a seguir.*

|SALÁRIO|CLASSIFICAÇÃO|
|---|---|
|Até  700,00 (inclusive)|Mal remunerado|
|Maiores que  700,00|Bem remunerado|


In [ ]:
print('Menu de opções:')
print('1 - Imposto')
print('2 - Novo salário')
print('3 - Classificação')
opc = int(input('Digite a opção desejada: '))
if opc == 1:
    salario = int(input('Informe o salário: R$'))
    if salario < 500:
        imposto = salario * 0.05
        print(F'O imposto para o salário informado é de R${imposto:.2f}')
    elif salario >= 500 and salario <= 850:
        imposto = salario * 0.1
        print(F'O imposto para o salário informado é de R${imposto:.2f}')
    else:
        imposto = salario * 0.15
        print(F'O imposto para o salário informado é de R${imposto:.2f}')
if opc == 2:
    salario = int(input('Informe o salário: R$'))
    if salario >1500:
        nsalario = salario + 25
        print(F'O novo salario é de R${nsalario}')
    elif salario >= 750 and salario <= 1500:
        nsalario = salario + 50
        print(F'O novo salario é de R${nsalario}')
    elif salario >= 450 and salario < 750:
        nsalario = salario + 75
        print(F'O novo salario é de R${nsalario}')
    elif salario < 450:
        nsalario = salario + 100
        print(F'O novo salario é de R${nsalario}')
if opc == 3:
    salario = int(input('Informe o salário: R$'))
    if salario <= 700:
        print('Você é mal remunerado.')
    else:
        print('Você é bem remunerado.')

Menu de opções:
1 - Imposto
2 - Novo salário
3 - Classificação
Digite a opção desejada: 1
Informe o salário: R$1830
O imposto para o salário informado é de R$274.50


###25. Faça um programa que receba o salário inicial de um funcionário, calcule e mostre o novo salário, acrescido de bonificação e de auxílio escola.

|SALÁRIO|BONIFICAÇÃO|
|---|---|
|Até  500,00|5% do salário|
|Entre  500,00 e 1.200,00|12% do salário|
|Acima de  1.200,00|Sem bonificação|
>
|SALÁRIO|AUXÍLIO ESCOLA|
|---|---|
|Até  600,00| 150,00|
|Acima de  600,00| 100,00|

novo_salario = salario + bonificacao + auxílio escola

In [ ]:
salario = int(input('Informe o salário:'))
if salario > 1200:
    nsalario = salario + 100
    print(F'O seu novo salário é: R${nsalario}')
if salario <= 600:
    if salario < 500:
        nsalario = salario + (salario * 0.05) + 150
        print(F'Seu novo salário é R${nsalario}')
    nsalario = salario + 150
    print(F'Seu novo salário é R${nsalario}')
elif salario > 600:
    if salario >= 500 and salario <= 1200:
        nsalario = salario + (salario * 0.12) + 100
        print(F'Seu novo salário é R${nsalario}')
    nsalario = salario + 100
    print(F'Seu novo salário é R${nsalario}')

Informe o salário:1100
Seu novo salário é R$1332.0
Seu novo salário é R$1200


### 26. Faça um programa que receba o valor do salário mínimo, o número de horas trabalhadas, o número de dependentes do funcionário e a quantidade de horas extras trabalhadas. Calcule e mostre o salário a receber do funcionário de acordo com as regras a seguir:

  * O valor da hora trabalhada é igual a 1/5 do salário mínimo.
  * O salário do mês é igual ao número de horas trabalhadas multiplicado pelo valor da hora trabalhada.
  * Para cada dependente, acrescentar  32,00.
  * Para cada hora extra trabalhada, calcular o valor da hora * trabalhada acrescida de 50%.
  * O salário bruto é igual ao salário do mês mais o valor dos dependentes mais o valor das horas extras.
  * Calcular o valor do impostoosto de renda retido na fonte de acordo com a tabela a seguir:

|IRFF|SALÁRIO BRUTO|
|---|---|
|Isento|Inferior a  200,00|
|10%|De  200,00 até 500,00|
|20%|Superior a  500,00|

  * O salário líquido é igual ao salário bruto menos IRRF.
  * A gratificação é de acordo com a tabela a seguir:

|SALÁRIO LÍQUIDO|GRATIFICAÇÃO|
|---|---|
|Até 350,00| 100,00|
|Superior a  350| 50,00|

  * O salário a receber do funcionário é igual ao salário líquido mais a gratificação.

  ```code
  LEIA salario_minimo, numero_horas_trabalhadas, numero_dependentes, numero_horas_extras
  
  valor_hora = 1/5 * salario_minimo
  salario_mes = numero_horas_trabalhadas * valor_hora
  valor_dependentes = 32 * numero_dependentes
  valor_hora_extra = numero_horas_extras * (valor_hora + (valor_hora * 50/100))
  salario_bruto =salario_mes + valor_dependentes + valor_hora_extra
  
  SE salario_bruto < 200
        imposto = 0
  SENÃO SE salario_bruto >= 200 E salario_bruto <= 500
        imposto = salario_bruto * 10/100
  SENÃO SE salario_bruto > 500
        imposto = salario_bruto * 20/100
  salario_liquido = salario_bruto – imposto
  SE salario_liquido <= 350
        gratificacao = 100
  SENÃO
        gratificacao = 50
  salario_a_receber = salario_liquido + gratificacao
  
  ESCREVA salario_a_receber
  ```

In [ ]:
# Digite seu código aqui

### 27 Um supermercado deseja reajustar os preços de seus produtos usando o seguinte critério: o produto poderá ter seu preço aumentado ou diminuído. Para o preço ser alterado, o produto deve preencher pelo menos um dos requisitos a seguir:

|VENDA MÉDIA MENSAL|PREÇO ATUAL|% DE AUMENTO|% DE DIMINUIÇÃO|
|---|--|---|---|
|< 500|<  30000|10|-|
|\>= 500 e <1200|\>=  30.00 e <  80.00|15|-|
|\>= 1200|\>=  80.00|-|20|

* Faça um programa que receba o preço atual e a venda média mensal do produto, calcule e mostre o novo preço.

```code
LEIA preco_atual, media_mensal_vendas

SE media_mensal_vendas < 500 OU preco_atual < 30
    novo_preco = preco_atual + 10/100 * preco_atual
SENÃO SE media_mensal_vendas >= 500 E media_mensal_vendas < 1200 OU preco_atual >= 30 E preco_atual<80
    novo_preco = preco_atual + 15 / 100 * preco_atual
SENÃO SE venda >= 1200 OU preco_atual >= 80
    novo_preco = preco_atual – 20 / 100 * preco_atual
      
ESCREVA novo_preco
```

In [ ]:
# Digite seu código aqui

### 28. Faça um programa para resolver equações do 2 o grau.

**ax² + bx + c = 0**

A variável *a* deve ser diferente de zero*

  * delta = b ** 2 - 4 * a * c
  * delta < 0.  **Não existe raiz real**
  * delta = 0. **Existe uma raiz real**
    * x = (-b) / (2 * a)
  * delta > 0. **Existem duas raízes reais**
    * x1 = -b + raiz(delta)/ (2 * a)
    * x2 = -b  - raiz(delta)/ (2 * a)

```code
LEIA a, b, c
SE a = 0
    ESCREVA “Estes valores não formam uma equação de segundo grau” 
SENÃO
    delta = (b * b) – ( 4 * a * c)
    SE delta < 0
        ESCREVA “Não existe raiz real”
    SE delta = 0
        ESCREVA “Existe uma raiz real”
        x1 = (– b) / (2 * a)
        ESCREVA x1
    SE delta > 0
        ESCREVA “Existem duas raízes reais”
        x1 = (– b) + raiz(delta) / (2 * a)
        x2 = (– b) - raiz(delta) / (2 * a)
        ESCREVA(x1, x2)
```

In [ ]:
# Digite seu código aqui

### 29. Dados três valores X, Y e Z, verifique se eles podem ser os comprimentos dos lados de um triângulo e, se forem, verifique se é um triângulo equilátero, isósceles ou escaleno. Se eles não formarem um triângulo, escreva uma mensagem.
Considere que:
* o comprimento de cada lado de um triângulo é menor que a soma dos outros dois lados;
* chama-se equilátero o triângulo que tem três lados iguais;
* denomina-se isósceles o triângulo que tem o comprimento de dois lados iguais;
* recebe o nome de escaleno o triângulo que tem os três lados diferentes.

```code
LEIA x, y, z
SE x < y + z E y < x + z E z < x + y
    SE x = y E y = z
        ESCREVA “Triângulo Equilátero”
    SENÃO SE x = y OU x = z OU y = z
        ESCREVA “Triângulo Isósceles”
    SENÃO SE x ≠ y E x ≠ z E y ≠ z
        ESCREVA “Triângulo Escaleno”
SENÃO
    ESCREVA “Essas medidas não formam um triângulo”
```

In [ ]:
# Digite seu código aqui

### 30. Faça um programa que receba a altura e peso de uma pessoa. De acordo com a tabela a seguir, verifique e mostre a classificação dessa pessoa.

|ALTURA|ATÉ 60|ENTRE 60 E 90 (INCLUSIVE)|ACIMA DE 90|
|---|---|---|---|
|Menores que 1,20|A|D|G|
|De 1,20 a 1,70|B|E|H|
|Maiores que 1,70|C|F|I|

```code
LEIA altura, peso
SE altura < 1.20
    SE peso <= 60
        ESCREVA “A”
    SENÃO SE peso > 60 E peso <= 90
        ESCREVA “D”
    SENÃO SE peso > 90
        ESCREVA “G”
SENÃO SE altura >= 1.20 E altura <= 1.70
    SE peso <= 60
        ESCREVA “B”
    SENÃO SE peso > 60 E peso <= 90
        ESCREVA “E”
    SENÃO SE peso > 90
        ESCREVA “H”
SENÃO SE altura > 1.70
    SE peso <= 60
        ESCREVA “C”
    SENÃO SE peso > 60 E peso <= 90
        EESCREVA “F”
    SENÃO SE peso > 90
        ESCREVA “I”
```

In [ ]:
# Digite seu código aqui

### 31. Faça um programa que receba:

* O código de um produto comprado, supondo que a digitação do código do produto seja sempre válida, isto é, um número * inteiro entre 1 e 10.
* O peso do produto em quilos.
* O código do país de origem, supondo que a digitação do  código seja sempre válida, isto é, um número inteiro entre 1 e 3.

|CÓDIGO DO PAÍS DE ORIGEM|IMPOSTO|
|---|---|
|1|0%|
|2|15%|
|3|25%|

|CÓDIGO DO PAÍS DO PRODUTO|PREÇO POR GRAMA|
|---|---|
|1 a 4|10|
|5 a 7|25|
|8 a 10|35|

Calcule e mostre:
* o peso do produto convertido em gramas;
* o preço total do produto comprado;
* valor do imposto, sabendo que ele é cobrado sobre o * preço total do produto comprado e dependendo país de origem;
* o valor total, preço total do produto mais imposto.

```code
LEIA codigo_produto, peso_quilos, codigo_pais
peso_em_gramas = peso_quilos * 1000
ESCREVA peso_em_gramas
SE codigo_produto >= 1 E codigo_produto <= 4
   preco_por_grama = 10
SENÃO SE codigo_produto >= 5 E codigo_produto <= 7
   preco_por_grama = 25
SENÃO SE codigo_produto >= 8 E codigo_produto <= 10
   preco_por_grama = 35    
preco_total = peso_em_gramas *preco_por_grama
ESCREVA preco_total
SE codigo_pais = 1
    imposto = 0
SENÃO SE codigo_pais = 2
    imposto = preco_total * 15/100
SENÃO SE codigo_pais = 3
    imposto = preco_total * 25/100    
ESCREVA imposto
valor_total = preco_total + imposto
ESCREVA valor_total
```

In [ ]:
# Digite seu código aqui

### 32. Faça um programa que receba:
* o código do estado de origem da carga de um caminhão, supondo que a digitação do código do estado seja sempre válida, isto é, um número inteiro entre 1 e 5;
* o peso da carga do caminhão em toneladas;
* o código da carga, supondo que a digitação do código seja sempre válida, isto é, um número inteiro entre 10 e 40.

| CÓDIGO DO ESTADO| IMPOSTO|
|---|---|
|1|35%|
|2|25%|
|3|15%|
|4|5%|
|5|Isento|
>
CÓDIGO DA CARGA|PREÇO POR QUILO
---|---
10 a 20|100
21 a 30|250
31 a 40|400

Calcule e mostre:
* o peso da carga do caminhão convertido em quilos;
* o preço da carga do caminhão;
* o valor do imposto, sabendo que o imposto é cobrado sobre o preço da carga do caminhão e depende do estado de origem;
* o valor total transportado pelo caminhão, preço da carga mais imposto.

```
LEIA codigo_estado, peso_em_toneladas, codigo_carga
peso_em_quilos = peso_em_toneladas * 1000
ESCREVA peso_em_quilos
SE codigo_carga >= 10 E codigo_carga <= 20
    preco_da_carga = 100 * peso_em_quilos
SENÃO SE codigo_carga >= 21 E codigo_carga <= 30
    preco_da_carga = 250 * peso_em_quilos
SENÃO SE codigo_carga >= 31 E codigo_carga <= 40
    preco_da_carga = 340 * peso_em_quilos   
ESCREVA preco_da_carga
SE codigo_estado = 1
    imposto = 35/100 * preco_da_carga
SENÃO SE codigo_estado = 2
    imposto = 25/100 * preco_da_carga
SENÃO SE codigo_estado = 3
    imposto = 15/100 * preco_ds_carga
SENÃO SE codigo_estado = 4
    imposto = 5/100 * preco_da_carga
SENÃO SE codigo_estado = 5
    imposto = 0
ESCREVA imposto
valor_total = preco_da_carga + imposto
ESCREVA valor_total
```

In [ ]:
# Digite seu código aqui

### 33. Faça um programa que receba o salário base e o tempo de serviço de um funcionário. Calcule e mostre:
* O imposto, conforme a tabela a seguir.

|SALÁRIO BASE|% SOBRE O SALÁRIO BASE|
|---|---|
|< 200|isento|
|>= 250 E <= 450|3%|
|>  450 E < 700|8%|
|>=  700|12%|

* A gratificação, de acordo com a tabela a seguir.

SALÁRIO BASE R$|TEMPO DE SERVIÇO|GRATIFICAÇÃO
---|---|---
Superior a 500,00|Até 3 anos|20
 | Mais de 3 anos|30
  --------------------------|---------|--------------
Até 500,00|Até 3 anos|23
|Entre 3 e 6 anos|35
|Acima de 6 anos|33

* O salário líquido, ou seja, salário base menos imposto mais gratificação.
* A categoria, que está na tabela a seguir.

```
LEIA salario_base, tempo

SE salario_base < 200
    imposto = 0
SENÃO SE salario_base <= 450
    imposto = 3/100 * salario_base
SENÃO SE sal_base < 700
    imposto = 8/100 * salario_base
SENÃO
    imposto = 12/100 * salario_base
ESCREVA imposto
SE salario_base > 500
    SE tempo <= 3
         gratificacao = 20
    SENÃO 
        gratificacao = 30
SENÃO
    SE tempo <= 3
        gratificacao = 23
    SENÃO SE tempo < 6
        gratificacao = 35
    SENÃO gratificacao = 33
ESCREVA gratificacao
salario_liquido = salario_base – imposto + gratificacao
ESCREVA salario_liqquido
SE salario_liquido <= 350
    ESCREVA “Classificação A”
SENÃO SE salario_liqquido < 600
    ESCREVA “Classificação B”
SENÃO
    ESCREVA “Classificação C”
```

In [ ]:
# Digite seu código aqui

### 34. Faça um programa que:
* receba o valor do salário mínimo,
* o turno de trabalho (M — matutino; V — vespertino; ou N — noturno),
* a categoria (O — operário; G — gerente) 
* número de horas trabalhadas no mês de um funcionário. 

*Suponha a digitação apenas de dados válidos e, quando houver digitação de letras, utilize maiúsculas.*

**Calcule e mostre:**

* O coeficiente do salário, de acordo com a tabela a seguir.

TURNO DE TRABALHO|VALOR DO COEFICIENTE
---|---
M - Matutino|10% do salário mínimo
V - Vespertino|15% do salário mínimo
N - Noturno|20% do salário mínimo

* O valor do salário bruto, ou seja, o número de horas trabalhadas multiplicado pelo valor do coeficiente do salário.

* O imposto, de acordo com a tabela a seguir.

CATEGORIA| SALÁRIO BRUTO|IMPOSTO SOBRE O SALÁRIO BRUTO
---|---|---
O - Operário|>= 300,00|5%
O - Operário|< 300,00|3%
G - Gerente|>= 300,00|6%
G - Gerente|< 300,00|4%

* A gratificação, de acordo com as regras a seguir. Se o funcionário preencher todos os requisitos a seguir, sua gratificação será de 50,00; caso contrário, será de 30,00. Os requisitos são:
 * Turno: Noturno
  * Número de horas trabalhadas: Superior a 80 horas
  * O auxílio alimentação, de acordo com as seguintes regras.
* Auxilio alimentação, um terço do seu salário bruto; caso contrário, será de metade do seu salário bruto. Os requisitos são:
  * Se o funcionário preencher algum dos requisitos a seguir, seu auxílio alimentação será de 
  * Categoria: Operário
  * Coeficiente do salário: < = 25
* O salário líquido, ou seja, salário bruto menos imposto mais gratificação mais auxílio alimentação.
* A classificação, de acordo com a tabela a seguir:

SALÁRIO LÍQUIDO|MENSAGEM
---|---
Menor que 350,00|Mal remunerado
Entre 350 e 600,00|Normal
Maior que 600,00|Bem remunerado

```
LEIA salario_minino, turno, categoria, numero_de_horas_trabalhadas
SE turno = “M”
     coeficiente = 10/100 * salario_minino
SENÃO SE turno = “V”
     coeficiente = 15/100 * salario_minino
SENÃO SE turno = “N”
     coeficiente = 12/100 * salario_minino
     
ESCREVA coeficiente

salario_bruto = numero_de_horas_trabalhadas * coeficiente

ESCREVA salario_bruto

SE categoria = “O”
    SE sal_bruto >= 300
        imposto = 5/100 * sal_bruto
    SENÃO
        imposto = 3/100 * sal_bruto
SENÃO 
    SE salario_bruto >= 400
        imposto = 6/100 * salario_bruto
    SENÃO
         imposto = 4/100 * salario_bruto
         
ESCREVA imposto

SE turno = “N” E numero_de_horas_trabalhadas > 80
    gratificacao = 50
SENÃO
    gratificacao = 30
    
ESCREVA gratificacao

SE categoria = “O” OU coeficiente <= 25
    auxilio = 1/3 * salario_bruto
SENÃO
    auxilio = 1/2 * salario_bruto
    
ESCREVA auxilio

salario_liquido = salario_bruto – imposto + gratificacao + auxilio

ESCREVA salario_liquido

SE salario_liquido < 350
     ESCREVA “Mal Remunerado”
SENÃO SE salario_liquido >= 350 E salario_liqquido <= 600
     ESCREVA “Normal”
SENÃO SE salario_liquido > 600
     ESCREVA “Bem Remunerado”
```

In [ ]:
# Digite seu código aqui

### 35. Faça um programa que receba de um produto:
* o preço
* o tipo (A — alimentação; L — limpeza; e V — vestuário)
* a refrigeração (S — produto que necessita de refrigeração; e N — produto que não necessita de refrigeração) .

**Suponha que haverá apenas a digitação de dados válidos e, quando houver digitação de letras, utilize maiúsculas. Calcule e mostre:**

* O valor adicional, de acordo com a tabela a seguir:

REFRIGERAÇÃO|TIPO|PREÇO|VALOR ADICIONAL
---|---|---|---
N|A|<  15,00|2,00|
 | |>= 15,00|5,00|
 |---------|--------------|-------------------------------
 |L|< 10,00|1,50
 | |>= 10,00|2,50
  |---------|--------------|-------------------------------
 |V|< 30,00|3,00
 ||>= 30,00|2,50
  --------------------------|---------|--------------|-------------------------------
S|A||8,00
|L||0,00
|V||0,00

* O valor do imposto, de acordo com a regra a seguir.

PREÇO|PERCENTUAL SOBRE O PREÇO
---|---
< 25,00|5%
>= 25,00|8%

O preço de custo, ou seja, preço mais imposto.
O desconto, de acordo com a regra a seguir.

O produto que não preencher nenhum dos requisitos a seguir terá desconto de 3%, caso contrário, 0 (zero). Os requisitos são:
* Tipo: A
* Refrigeração: S

* O novo preço, ou seja, preço de custo mais adicional menos desconto.
* A classificação, de acordo com a regra a seguir.

NOVO PREÇO|CLASSIFICAÇÃO
---|---
<= 50,00|Barato
Entre 50,00 e 100,00|Normal
>= 100,00|Caro


```
LEIA pre, tipo, refrig
SE refrig = “N”
    SE tipo = “A”
        SE pre < 15
           valor_adic = 2
        SENÃO 
           valor_adic = 5
    SE tipo = “L”
        SE pre < 10
           valor_adic = 1.5
        SENÃO 
           valor_adic = 2.5
    SE tipo = “V”
        SE pre < 30
           valor_adic = 3
        SENÃO 
           valor_adic = 2.5
SENÃO SE tipo = “A”
    valor_adic = 8
    SE tipo = “L”
        valor_adic = 0
    SE tipo = “V”
        valor_adic = 0
ESCREVA valor_adic
SE pre < 25
    imposto = 5/100 * pré
SENÃO 
    imposto = 8/100 * pre
ESCREVA imposto
pre_custo = pre + imposto
ESCREVA pre_custo
SE tipo != “A” E refrig != “S”
  desconto = 3/100 * pre_custo
SENÃO 
  desconto = 0
ESCREVA desconto
novo_pre = pre_custo + valor_adic – desconto
ESCREVA novo_pre
SE novo_pre <= 50
    ESCREVA “Barato”
SENÃO SE novo_pre < 100
    ESCREVA “Normal”
SENÃO 
    ESCREVA “Caro”


In [ ]:
# Digite seu código aqui

### 36. Faça um programa que receba a medida de um ângulo em graus. Calcule e mostre o quadrante em que se localiza esse ângulo. Considere os quadrantes da trigonometria e, para ângulos maiores que 360° ou menores que −360°, reduzí-los, mostrando também o número de voltas e o sentido da volta (horário ou anti-horário).
```
SE angulo > 360 OU angulo < -360
    voltas = parte inteira(angulo / 360)    
    angulo = RESTO(angulo / 360)
SENÃO 
    voltas = 0
SE angulo = 0 OU angulo = 90 OU angulo = 180
           OU angulo = 270 OU angulo = 360
           OU angulo = -90 OU angulo = -180
           OU angulo = -270 OU angulo = -360
  ESCREVA “Está em cima de algum dos eixos”
  SE (angulo > 0 E angulo < 90) OU (angulo < -270 E angulo > -360)
    ESCREVA “1o Quadrante”
  SENÃO SE (angulo > 90 E angulo < 180) OU (angulo < -180 E angulo > -270)
    ESCREVA “2o Quadrante”
  SENÃO SE (angulo > 180 E angulo < 270) OU (angulo < -90 E angulo > -180)
    ESCREVA “3o Quadrante”
  SENÃO SE (angulo > 270 E angulo < 360) OU (angulo < 0 E angulo > -90)
    ESCREVA “4o Quadrante”
  ESCREVA voltas, “ volta(s) no sentido “
SE angulo < 0
    ESCREVA “horário”
SENÃO 
    ESCREVA “anti-horário”
```

In [ ]:
# Digite seu código aqui